dev/com.ibm.ws.jsf.2.2_fat.2/fat/src/com/ibm/ws/jsf22/fat/tests/JSF22MiscellaneousTests.java

```diff
@@ -657,4 +663,43 @@ public void testMyFaces4512() throws Exception {
             assertTrue("The MyFaces4512ViewHandler was not invoked!", !jsf22MiscellaneousServer.waitForStringInLog("MyFaces4512ViewHandler was invoked!").isEmpty());
         }
     }
+    /*
+     * All the XML should be valid by default. 
+     */
+    @SkipForRepeat({NO_MODIFICATION,EE8_FEATURES,EE10_FEATURES})
+    @Test
+    public void testXMLValidationGoodCase() throws Exception {
+
+        jsf22MiscellaneousServer.setMarkToEndOfLog();
+
+        ShrinkHelper.defaultDropinApp(jsf22MiscellaneousServer, APP_NAME_XML_VALIDATION_GOOD + ".war");
+
+        assertTrue("XML Validation Error found erronously.", jsf22MiscellaneousServer.findStringsInLogsAndTraceUsingMark(".*cvc-complex-type.2.4.a: Invalid content was found starting with element.*").isEmpty()); 
+
+        assertTrue("XML Validation Error found erronously.", jsf22MiscellaneousServer.findStringsInLogsAndTraceUsingMark(".*org.xml.sax.SAXParseException.*").isEmpty());
+
+        jsf22MiscellaneousServer.resetLogMarks();
+
+        jsf22MiscellaneousServer.removeAndStopDropinsApplications(APP_NAME_XML_VALIDATION_GOOD  + ".war");
+    }
+
+    /*
+     * The bad case involves the faces-config.xml including an unexpected element. Validation should fail with following exception:
+     * [ERROR   ] cvc-complex-type.2.4.a: Invalid content was found starting with element '{"http://xmlns.jcp.org/xml/ns/javaee":wrong-element}'.
+     */
+    @SkipForRepeat({NO_MODIFICATION,EE8_FEATURES})
```

Comment:
so the validation works with EE9/10/11? but doesn't for EE7/EEE8?

Reply:
Other errors occurs. See my comment here: https://github.com/OpenLiberty/open-liberty/pull/30500#issuecomment-2585040446